### ISSUE TITLE
Programatically loading Webhook Relay node 

### ISSUE THREAD

**Issue by geoffatblacktusk** [webhookrelay/node-red-contrib-webhookrelay#4]
We have a middleware application that we are attempting to configure entirely from external config files.
This is because it is intended to be deployed at different customer sites, and we do not wish to edit the NodeRed flows for each deployment.

This has largely affected the http authorisation methods and keys for the various APIs that we need to initialise, but if we use Webhook relay it will also affect the Key/Secret pair.

It seems that as there is no input allowed for this node, that there is no way to pass in the connection parameters?

**Comment by rusenask** [comment]
Hi @geoffatblacktusk can you give me an example of some other node how they do this kind of configuration that you have in mind? 
As I understand, you would want to call Node-RED API directly to configure Webhook Relay key/secret pair and bucket configuration?

**Comment by geoffatblacktusk** [comment]
Thankyou for the reply, I was thinking along the lines of the http request node, where the authorisation can be set on the node or passed in via msg.headers.
I was not planning to use the API, just setting up a startup inject on the flow..

**Comment by rusenask** [comment]
okay, so basically something like that (just can't connect them now because there's no input in the whr node):

![image](https://user-images.githubusercontent.com/9080105/70475040-9f35ae00-1acb-11ea-90e3-9740012b8cd1.png)

And the input could be something like:

```
{
  "action": "subscribe",
  "buckets": ["foo"],
  "key": "xx",
  "secret": "yy"
}
```

Would this work?

**Comment by geoffatblacktusk** [comment]
Yes – exactly, obviously you would want to make the input optional

Regards
Geoff

[Blacktusk HTML (Geoff-3).bmp]
